File: submissions/code-freeze-is-coming/10.js
function f(e,r){function n(e,r,n){return e.substr(0,r)+n+e.substr(r+n.length)}for(var l=r.split("\n").reverse(),a=[],s="",f=0,t=0;t<l[0].length;t++)if("X"===l[0][t])for(var i=0,c=!0,o=1;o<l.length;o++){switch(l[o][t]){case"|":break;case"/":1>i?i++:c=!1;break;case"\\":i>-1?i--:c=!1;break;case".":0===i?(a.push({line:o,column:t}),f++):c=!1;break;case"$":0===i?(e--,f++):c=!1;break;default:c=!1}if(!c)break}if(f)if(0>e)s=e;else if(e>a.length)s=e-a.length;else for(var u=0;e>u;u++){var b=a[u];l[b.line]=n(l[b.line],b.column,"$")}else s=":(";if(!s){s=l[0];for(var h=1;h<l.length;h++)s=l[h]+"\n"+s}console.log(s)}

File: submissions/igyuk-meg-a-compareto-t/10.rb

def f(q,s)
    s=s.split("\n").map{|x|x.split('')}.transpose
    w=[]
    f=[]
    s.each_with_index{|q, y|
        t = q.reverse
        sz = t.size - 2
        next unless t[0]=='X'
        hz=true
        zz=false
        z=0
        t.drop(1).each_with_index{|i, x|
            if hz && zz && z==0 then
                w << [sz-x,y] if i=='.'
                f << [sz-x,y] if i=='$'
            end
            zz=true if '/|\\'.index i
            hz=false unless '/|\\'.index i
            z+=1 if i=='/'
            z-=1 if i=='\\'
        }
    }
    
    l = f.count
    d = q - l
    
    if f.count + w.count == 0 then
        puts ':('
    elsif d < 0 then
        puts d
    elsif d == 0 then
        puts 0
    elsif d <= w.count then
        while d > 0 do
            c=w.shift
            s[c[1]][c[0]]='$'
            d-=1
        end
        
        puts s.transpose.map{|x|x.join('')}.join("\n")
    else
        puts d - w.count
    end
end